roles/bootstrap-os/tasks/bootstrap-ubuntu.yml

```diff
@@ -9,36 +7,57 @@
       - python-pip
       - dbus
 
-- name: Bootstrap | Check if bootstrap is needed
-  raw: dpkg -l | cut -d' ' -f3 |grep -e ^{{item}}$
+- name: Check if bootstrap is needed
+  raw: dpkg -l | cut -d' ' -f3 | grep -e ^{{ item }}$
```

Comment:
All other OS-specific plays use `which ...` for this, Ubuntu is the odd one out here.

Reply:
which is to locate a command not a package. some packages don't install any binaries

Reply:
and `which python-apt`  would fail

Reply:
Yes, but the `apt` module can deal with a missing `python-apt` package by itself.

Maybe however on most other distros also the locally installed packages should be queried instead of using `which ...` in that case?

Reply:
It is not something I would change right now, just something that I found odd, considering that even the Debian one does it differently.

Reply:
Looking at blame, I made the change. If I recall it was because the bootstrap would ALWAYS trigger since "which python-apt" was always failing. The problem is probably the same in debian&co, so we should fix them too at some point (not in this PR)

Reply:
The debian one doesn't check for or install python-apt at all.